tests/priority-queue.spec.ts

```diff
@@ -0,0 +1,599 @@
+			expect(instance.getChildNodesIndexes(8)).toStrictEqual({left: null, right: null});
+		});
+	});
+
+	describe('getNextIndex', () => {
+		beforeEach(() => {
+			items.forEach((item) => {
+				instance.push(item);
+			});
+		});
+
+		it('should call getParentNodeIndex and not call getChildNodesIndexes if startFromTop is false', () => {
+			const spyGPNI = jest.spyOn(instance, 'getParentNodeIndex');
+			const spyGCNI = jest.spyOn(instance, 'getChildNodesIndexes');
+			spyGPNI.mockClear();
+			spyGCNI.mockClear();
+			instance.getNextIndex(false, 1);
+			expect(spyGPNI).toBeCalledTimes(1);
+			expect(spyGCNI).toBeCalledTimes(0);
+		});
+
+		it('should call getChildNodesIndexes and not call getParentNodeIndex if startFromTop is true', () => {
+			const spyGPNI = jest.spyOn(instance, 'getParentNodeIndex');
+			const spyGCNI = jest.spyOn(instance, 'getChildNodesIndexes');
+			spyGPNI.mockClear();
+			spyGCNI.mockClear();
+			instance.getNextIndex(true, 1);
+			expect(spyGPNI).toBeCalledTimes(0);
+			expect(spyGCNI).toBeCalledTimes(1);
+		});
+
+		it('should return null if both child indexes are null', () => {
+			expect(instance.getNextIndex(true, 8)).toBe(null);
+		});
+
+		it('should return leftIndex if rightIndex is null', () => {
+			instance.push(100);
+			expect(instance.getNextIndex(true, 4)).toBe(9);
+		});
+
+		it('should return leftIndex if elements[rightIndex] is null', () => {
+			instance.elements[2] = null!;
+			expect(instance.getNextIndex(true, 0)).toBe(1);
+		});
+
+		it('should return rightIndex if elements[leftIndex] is null', () => {
+			instance.elements[1] = null!;
+			expect(instance.getNextIndex(true, 0)).toBe(2);
+		});
+
+		it('should return leftIndex if elements[leftIndex] < elements[rightIndex]', () => {
+			instance.elements[1] = 10;
+			instance.elements[2] = 20;
+			expect(instance.getNextIndex(true, 0)).toBe(1);
+		});
+
+		it('should return rightIndex if elements[leftIndex] > elements[rightIndex]', () => {
+			instance.elements[1] = 20;
+			instance.elements[2] = 10;
+			expect(instance.getNextIndex(true, 0)).toBe(2);
+		});
+	});
+
+	describe('isHeapUnbalanced', () => {
+		beforeEach(() => {
+			items.forEach((item) => {
+				instance.push(item);
+			});
+			instance.push(1);
+		});
+
+		it('should return false if either parameter is not a number', () => {
+			expect(instance.isHeapUnbalanced(null, null)).toBe(false);
+			expect(instance.isHeapUnbalanced(null, 0)).toBe(false);
+			expect(instance.isHeapUnbalanced(0, null)).toBe(false);
+		});
+
+		it('should return false if either node is null', () => {
+			instance.elements[0] = null!;
+			expect(instance.isHeapUnbalanced(0, 1)).toBe(false);
+			instance.elements[0] = 10;
+			instance.elements[1] = null!;
+			expect(instance.isHeapUnbalanced(0, 1)).toBe(false);
+			instance.elements[0] = null!;
+			expect(instance.isHeapUnbalanced(0, 1)).toBe(false);
+		});
+
+		it('should use the proper parameter order decided by startFromTop', () => {
```

Comment:
Test is okay, but change the word _proper_ to whatever proper should confer here.

If this was written to test a strictly ascending sorted list of numbers, I might change:
`should sort elements in proper order` 
to 
`should sort elements in ascending numerical order`.
Small difference, but assume the reader won't know what proper actually means.